ui/src/app/kvm/components/KVMHeaderForms/AddKVM/AddLxd/AddLxd.tsx

```diff
@@ -48,21 +51,38 @@ export const AddLxd = ({
     podSelectors.getProjectsByLxdServer(state, newPodValues.power_address)
   );
 
+  // Once a generated certificate for the new pod exists in state, the user
+  // should be sent to the auth trust step.
+  const shouldGoToAuthStep = !!(
+    generatedCertificate?.CN.includes(newPodValues.name) &&
+    step === AddLxdSteps.CREDENTIALS
+  );
+  useEffect(() => {
+    if (shouldGoToAuthStep) {
+      setStep(AddLxdSteps.AUTHENTICATION);
+    }
+  }, [shouldGoToAuthStep]);
+
   // User is considered "authenticated" if they have set a LXD server address
   // and projects for it exist in state. Once "authenticated", they should
   // be sent to the project selection step.
-  const authenticated = !!newPodValues.power_address && projects.length >= 1;
+  const shouldGoToProjectStep = !!(
+    newPodValues.power_address &&
+    projects.length >= 1 &&
+    step !== AddLxdSteps.SELECT_PROJECT
+  );
   useEffect(() => {
-    if (authenticated && step !== AddLxdSteps.SELECT_PROJECT) {
+    if (shouldGoToProjectStep) {
       setStep(AddLxdSteps.SELECT_PROJECT);
     }
-  }, [authenticated, step]);
+  }, [shouldGoToProjectStep]);
 
   // We run the cleanup function here rather than in each form component
   // because we want the errors to be able to persist across forms.
   useEffect(() => {
     return () => {
       dispatch(podActions.cleanup());
+      dispatch(generalActions.clearGeneratedCertificate());
```

Comment:
It might be helpful to leave a comment as to why this is necessary. Something like: "The generated certificate state only holds one certificate at a time so the state can be cleaned up before fetching another certificate", but I'm sure you can make it make more sense than that :).